test/k23si/3SITxnTest.cpp

```diff
@@ -1794,10 +1822,1277 @@ seastar::future<> testScenario02() {
+    K2INFO("+++++++ TestScenario 08: server-initiated txn abort +++++++");
+    
+    return seastar::make_ready_future()
+    .then([] {
+        return getTimeNow();
+    })
+    .then([this](dto::Timestamp&& ts) {
+        return seastar::do_with(
+            // multi MTRs are used for multiple transactions to execute. We use these MTRs in sequence
+            dto::K23SI_MTR { .txnid = txnids++, .timestamp = {(ts.tEndTSECount()), 123, 1000}, .priority = dto::TxnPriority::Medium },
+            dto::K23SI_MTR { .txnid = txnids++, .timestamp = {(ts.tEndTSECount() + 20000), 123, 1000}, .priority = dto::TxnPriority::Medium },
+            dto::K23SI_MTR { .txnid = txnids++, .timestamp = {(ts.tEndTSECount() + 30000), 123, 1000}, .priority = dto::TxnPriority::Medium },
+            dto::K23SI_MTR { .txnid = txnids++, .timestamp = {(ts.tEndTSECount() + 40000), 123, 1000}, .priority = dto::TxnPriority::Medium },
+            dto::K23SI_MTR { .txnid = txnids++, .timestamp = {(ts.tEndTSECount() + 50000), 123, 1000}, .priority = dto::TxnPriority::Medium },
+            dto::Key {.schemaName = "schema", .partitionKey = "SC08_pkey1", .rangeKey = "rKey1"},
+            DataRec {.f1="SC05_f1_zero", .f2="SC04_f2_zero"},
+            [this](auto& mtr1, auto& mtr2, auto& mtr3, auto& mtr4, auto& mtr5, auto& k1, auto& v0) {
+            K2INFO("------- SC08.case01 ( Txn with a old timestamp WI is PUSHed by another txn's READ ) -------");
+            return doWrite(k1, v0, mtr1, k1, collname, false, true, ErrorCaseOpt::NoInjection)
+            .then([](auto&& response)  {
+                auto& [status, val] = response;
+                K2INFO("SC08.prepare::OP_WI_old_k1. " << "status: " << status.code << " with MESG: " << status.message);
+                K2EXPECT(status, dto::K23SIStatus::Created);
+            })
+            .then([&] {
+                return doRead(k1, mtr2, collname, ErrorCaseOpt::NoInjection)
+                .then([](auto&& response)  {
+                    auto& [status, val] = response;
+                    K2INFO("SC08.cases01::OP_read_face_old_WI. " << "status: " << status.code << " with MESG: " << status.message);
+                    K2EXPECT(status, dto::K23SIStatus::KeyNotFound);
+                });
+            })
+            .then([&] {
+                return doInspectTxn(k1, mtr1, collname)
+                .then([&](auto&& response)  {
+                    auto& [status, val] = response;
+                    K2INFO("SC08.case01::OP_inspect_txn_1. " << "status: " << status.code << " with MESG: " << status.message);
+                    K2INFO("State of txn_1: " << val.state);
+                    K2EXPECT(status, dto::K23SIStatus::OK);
+                    K2EXPECT(val.state, dto::TxnRecordState::ForceAborted);
+                });
+            })
+            .then([&] {
+                K2INFO("------- SC08.case02 ( Txn WRITE happens before the Read-Cache record read time ) -------");
```

Comment:
You should do an END (with commit = false and syncFinalize = true) first to clean up the first write before doing this test case. Then you would expect AbortConflict on the response (as it is now) and InProgress on the txn inspect